generated/nifgen/nifgen/_grpc.py

```diff
@@ -0,0 +1,514 @@
+        response = self._invoke(
+            self._client.GetExtCalLastTemp,
+            grpc_types.GetExtCalLastTempRequest(vi=self._vi),
+        )
+        return response.temperature
+
+    def get_ext_cal_recommended_interval(self):  # noqa: N802
+        response = self._invoke(
+            self._client.GetExtCalRecommendedInterval,
+            grpc_types.GetExtCalRecommendedIntervalRequest(vi=self._vi),
+        )
+        return response.months
+
+    def get_hardware_state(self):  # noqa: N802
+        response = self._invoke(
+            self._client.GetHardwareState,
+            grpc_types.GetHardwareStateRequest(vi=self._vi),
+        )
+        return enums.HardwareState(response.state_raw)
+
+    def get_self_cal_last_date_and_time(self):  # noqa: N802
+        response = self._invoke(
+            self._client.GetSelfCalLastDateAndTime,
+            grpc_types.GetSelfCalLastDateAndTimeRequest(vi=self._vi),
+        )
+        return response.year, response.month, response.day, response.hour, response.minute
+
+    def get_self_cal_last_temp(self):  # noqa: N802
+        response = self._invoke(
+            self._client.GetSelfCalLastTemp,
+            grpc_types.GetSelfCalLastTempRequest(vi=self._vi),
+        )
+        return response.temperature
+
+    def get_self_cal_supported(self):  # noqa: N802
+        response = self._invoke(
+            self._client.GetSelfCalSupported,
+            grpc_types.GetSelfCalSupportedRequest(vi=self._vi),
+        )
+        return response.self_cal_supported
+
+    def import_attribute_configuration_buffer(self, configuration):  # noqa: N802
+        self._invoke(
+            self._client.ImportAttributeConfigurationBuffer,
+            grpc_types.ImportAttributeConfigurationBufferRequest(vi=self._vi, configuration=configuration),
+        )
+
+    def import_attribute_configuration_file(self, file_path):  # noqa: N802
+        self._invoke(
+            self._client.ImportAttributeConfigurationFile,
+            grpc_types.ImportAttributeConfigurationFileRequest(vi=self._vi, file_path=file_path),
+        )
+
+    def initialize_with_channels(self, resource_name, channel_name, reset_device, option_string):  # noqa: N802
+        response = self._invoke(
+            self._client.InitializeWithChannels,
+            grpc_types.InitializeWithChannelsRequest(resource_name=resource_name, channel_name=channel_name, reset_device=reset_device, option_string=option_string),
+        )
+        return response.vi
+
+    def initiate_generation(self):  # noqa: N802
+        self._invoke(
+            self._client.InitiateGeneration,
+            grpc_types.InitiateGenerationRequest(vi=self._vi),
+        )
+
+    def is_done(self):  # noqa: N802
+        response = self._invoke(
+            self._client.IsDone,
+            grpc_types.IsDoneRequest(vi=self._vi),
+        )
+        return response.done
+
+    def lock(self):  # noqa: N802
+        response = self._invoke(
```

Comment:
@DavidCurtiss why is the lock method for fgen not the local recursive lock?

Reply:
You are going to get errors (if not implemented by server) or deadlocks (if yes implemented) whenever you use this, right?

That's ok for now since we are still building this. Just wanting to understand.